databricks_connector.py

```diff
@@ -0,0 +1,715 @@
+# File: databricks_connector.py
+#
+# Copyright (c) 2023 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+
+import json
+import sys
+import traceback
+from datetime import datetime
+from pathlib import Path
+from typing import Optional
+
+import phantom.app as phantom
+import requests
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+
+import databricks_consts as consts
+
+deps_path = Path(__file__).parent.joinpath("dependencies")
```

Comment:
This is a rather bizarre bug related to dependency resolution in Python. 

On the SOAR Dev Container, `google.auth` fails to import if we don't do this. It's technically not needed in production, but it's harmless and allows us to test on the dev container.

We do a similar workaround in the Gmail connector.

If you want full context, there's an absurdly long Slack thread tracking it down: https://splunk.slack.com/archives/C02KYLKLQ02/p1715032637372529

Reply:
I agree that this isn't a good thing to have in place long-term. 

Since this is only needed for dev, and we're done developing on this connector for awhile after this release, I'm going to remove it for now. I have a fix in mind for the dev container that will make it unnecessary going forward, and I'm willing to gamble that we can get that in before anyone touches this connector again.

I'll note that the Gmail app's fix is even nastier (https://github.com/splunk-soar-connectors/googleworkspaceforgmail/blob/1578e77cac4f30cae9bbab264748a0c5545cbf10/gsgmail_connector.py#L40) and also doesn't even work anymore.